core/src/main/java/io/crate/monitor/SysInfoUtil.java

```diff
@@ -306,7 +306,8 @@ private void parseLsbVendor(File releaseFile) {
      * Taken from
      * https://github.com/hyperic/sigar/blob/master/src/os/linux/linux_sigar.c#L2585
      */
-    private void parseGenericVendor(File releaseFile) {
+    @VisibleForTesting
+    void parseGenericVendor(File releaseFile) {
```

Comment:
other minor suggestion (feel free to ignore): To improve testability the `line -> {}` method could be extracted. (So that a file is not necessary to test the extraction logical.